packages/terra-button-group/src/ButtonGroupButton.jsx

```diff
@@ -7,55 +7,121 @@ import styles from './ButtonGroupButton.scss';
+  ENTER: 13,
+  SPACE: 32,
+  TAB: 9,
+};
+
 const propTypes = {
   /**
-   * Indicates if the button should be selected on initial render.
+   * An optional icon. Nested inline with the text when provided
    */
-  isSelected: PropTypes.bool,
+  icon: PropTypes.element,
   /**
-   * Sets the button text
+   * Whether or not the button should be disabled.
    */
-  text: PropTypes.string,
+  isDisabled: PropTypes.bool,
   /**
-   * An optional icon. Nested inline with the text when provided
+   * Callback function triggered when button loses focus.
    */
-  icon: PropTypes.element,
+  onBlur: PropTypes.func,
   /**
    * Callback function triggered when clicked
    */
   onClick: PropTypes.func,
   /**
-   * Reverses the position of the icon and text
+   * Callback function triggered when button gains focus.
+   */
+  onFocus: PropTypes.func,
+  /**
+   * Callback function triggered when key is pressed.
+   */
+  onKeyDown: PropTypes.func,
+  /**
+   * Callback function triggered when key is released.
    */
-  isReversed: PropTypes.bool,
+  onKeyUp: PropTypes.func,
   /**
-   * Child Nodes
+   * Sets the button text. If an icon is provided, it will be an icon only button and this text is set as the aria-label for accessibility.
    */
-  children: PropTypes.node,
+  text: PropTypes.string.isRequired,
 };
 
 const defaultProps = {
-  isSelected: false,
-  isReversed: false,
+  isDisabled: false,
 };
 
+class ButtonGroupButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { focused: false };
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.handleKeyUp = this.handleKeyUp.bind(this);
+    this.handleOnBlur = this.handleOnBlur.bind(this);
+  }
 
-const ButtonGroupButton = ({ isSelected, ...customProps }) => {
-  const attributes = Object.assign({}, customProps);
+  handleOnBlur(event) {
+    this.setState({ focused: false });
 
-  const buttonClassName = cx([
-    'button-group-button',
-    { 'is-active': isSelected },
-    attributes.className,
-  ]);
-  attributes['aria-pressed'] = isSelected;
+    if (this.props.onBlur) {
+      this.props.onBlur(event);
+    }
+  }
 
-  return (
-    <Button
-      {...attributes}
-      className={buttonClassName}
-    />);
-};
+  handleKeyDown(event) {
+    // Add focus styles for keyboard navigation
+    if (event.nativeEvent.keyCode === KEYCODES.SPACE || event.nativeEvent.keyCode === KEYCODES.ENTER) {
+      this.setState({ focused: true });
+    }
+
+    if (this.props.onKeyDown) {
+      this.props.onKeyDown(event);
+    }
+  }
+
+  handleKeyUp(event) {
+    // Apply focus styles for keyboard navigation
+    if (event.nativeEvent.keyCode === KEYCODES.TAB) {
+      this.setState({ focused: true });
+    }
+
+    if (this.props.onKeyUp) {
+      this.props.onKeyUp(event);
+    }
+  }
+
+  render() {
+    const {
+      icon,
+      isDisabled,
+      ...customProps
+    } = this.props;
+
+    const attributes = Object.assign({}, customProps);
```

Comment:
General question - what's our reasoning behind using `Object.assign()` versus the spread operator `...customProps`?

Reply:
customProps would not be null if there is really no additional props so I don't think the reason is to ensure that we don't have an undefined customProps. There are components that have a need to add additional attributes. [Here](https://github.com/cerner/terra-core/blob/master/packages/terra-form-select/src/Select.jsx#L251) is an example. Perhaps, one reason to copy the custom props to another constant is to avoid adding additional attributes to the original customProps. But there may be other better reasons that I'm not aware of.

This line was already here before this change but I went ahead and removed it and just use customProps. https://github.com/cerner/terra-core/pull/1155/commits/5867f82edd6f7d3559befec0429fe8ee11183fa0